modules/javafx.graphics/src/main/native-glass/gtk/glass_window.cpp

```diff
@@ -702,10 +701,87 @@ WindowContextBase::~WindowContextBase() {
 }
 
 ////////////////////////////// WindowContextTop /////////////////////////////////
-WindowFrameExtents WindowContextTop::normal_extents = {28, 1, 1, 1};
-WindowFrameExtents WindowContextTop::utility_extents = {28, 1, 1, 1};
 
 
+// Work-around because frame extents are only obtained after window is shown.
+// This is used to know the total window size (content + decoration)
+// The first window will have a duplicated resize event, subsequent windows will use the cached value.
+WindowFrameExtents WindowContextTop::normal_extents = {0, 0, 0, 0};
+WindowFrameExtents WindowContextTop::utility_extents = {0, 0, 0, 0};
+
+static int geometry_get_window_width(const WindowGeometry *windowGeometry) {
+     return (windowGeometry->final_width.type != BOUNDSTYPE_WINDOW)
```

Comment:
minor: wouldn't it be more readable (here and below) if the test was reversed? ( `(windowGeometry->final_width.type != BOUNDSTYPE_WINDOW`)

Reply:
Those functions already existed. Do you prefer to keep as it was or to invert the `if`?